dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/CancellationTest.java

```
@@ -0,0 +1,140 @@
+        server.stopServer();
+    }
+
+    @Test
+    public void testCancellationToolWithCancellableParameterAndCancellationRequest() throws Exception {
+
+        final CountDownLatch latch = new CountDownLatch(1);
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+
+        Callable<String> threadCallingTool = () -> {
+            try {
+                String request = """
+                                  {
+                                  "jsonrpc": "2.0",
+                                  "id": "2",
+                                  "method": "tools/call",
+                                  "params": {
+                                    "name": "cancellationTool",
+                                    "arguments": {}
+                                  }
+                                }
+                                """;
+                //make sure this tread executes first
+                latch.countDown();
+                return HttpTestUtils.callMCP(server, "/toolTest", request);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        };
+
+        Future<String> future = executor.submit(threadCallingTool);
+
+        String cancellationRequestNotification = """
+                          {
+                          "jsonrpc": "2.0",
+                          "method": "notifications/cancelled",
+                          "params": {
+                            "requestId": "2",
+                            "reason": "no longer needed"
+                          }
+                        }
+                        """;
+        //make sure the tool call request has started
+        latch.await();
+        try {
+            //wait to make sure the tool call is registered before trying to cancel it
+            TimeUnit.MILLISECONDS.sleep(1000);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
```

Don't catch this, just let it fail the test if the sleep is interrupted.

---

Done